spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ldap/embedded/EmbeddedLdapAutoConfiguration.java

```diff
@@ -107,6 +109,22 @@ public InMemoryDirectoryServer directoryServer() throws LDAPException {
 					this.embeddedProperties.getCredential().getUsername(),
 					this.embeddedProperties.getCredential().getPassword());
 		}
+
+		if (!this.embeddedProperties.getValidation().isEnabled()) {
+			config.setSchema(null);
+		}
+		else if (this.embeddedProperties.getValidation().getSchema() != null) {
+			Resource schemaLocation = this.embeddedProperties.getValidation().getSchema();
+			try {
+				config.setSchema(Schema.mergeSchemas(Schema.getDefaultStandardSchema(),
+						Schema.getSchema(schemaLocation.getFile())));
```

Comment:
You can't really do that I am afraid. `Resource` is an abstraction and it may not return a file. The only reason why the tests currently pass is because you are running them with `target/test-classes` so resources effectively point to a file. If you use the same inside a repackaged jar, `getFile` will return `null`.

I am not super keen to introduce a feature that requires to point to `java.io.File`. Looking at the code for `Schema`, there isn't an easy way to read a `Schema` from a `InputStream`. IMO that feature should be available before we proceed with this PR